packages/x-components/src/utils/html.ts

```diff
@@ -10,3 +10,21 @@
 export function isElementEqualOrContained(a: Element, b: Element): boolean {
   return a === b || a.contains(b);
 }
+
+/**
+ * Returns the target element for a given event. The target element is got form `composedPath` Event
+ * method because if the event is triggered inside a Shadow DOM context, `event.target` points to
+ * Shadow DOM element, not the element that triggered the event. `composedPath` method also is
+ * available in a non-shadow DOM context. `event.target` is added as fallback.
+ *
+ * @remarks In a shadow DOM context, this function only works if the Shadow DOM uses `open`
+ * encapsulation mode.
+ *
+ * @param event - Event which takes place in the DOM.
+ * @returns Target Element of the event.
+ *
+ * @public
+ */
+export function getTargetElement(event: Event): Element {
+  return (event.composedPath()[0] ?? event.target) as Element;
```

Comment:
I kept `event.target` as fallback for retrocompatibility, but it wouldn't be needed because `event.composedPath()[0]` should always have the target element. WDTY? Can we get rid of it?

Reply:
AFAIK there is no possible way that the `composedPath` method returns an empty array, so I would get rid of the `event.target`.

Reply:
Removed then!